webapp/apps/taxbrain/views.py

```diff
@@ -130,198 +137,337 @@ def normalize(x):
     ans = [i.split('#') for i in x]
     return ans
 
-def process_model(model, start_year, stored_errors=None, request=None,
-                  do_full_calc=True, user=None):
+
+def parse_errors_warnings(errors_warnings, map_back_to_tb):
     """
-    take data from the model and submit the microsimulation job
-    inputs:
-        model: a TaxSaveInputs model instance
-        stored_errors: a dict of errors from the form or None
-        request: a Django request object, or None
-        do_full_calc: bool, if True, do the full calculation
-        user: instance of User model or None
-    returns:
-        unique_url: OutputUrl model instance
+    Parse error messages so that they can be mapped to Taxbrain param ID. This
+    allows the message to be displayed under the field where the value is
+    entered.
+
+    TaxCalculator returns an error message for all fields including those
+    specified implicitly by wildcards or by not entering a number for each
+    year in the remaining budget window.  This function keeps only the error
+    messages where the user explicitly specified a number for a budget window.
```

Comment:
I need to update this comment.  Initially, I planned to only show one error message.  However, we decided that for now all error messages will be displayed.